source/slang/slang-ir-bit-field-accessors.cpp

```diff
@@ -69,16 +65,42 @@ static void synthesizeBitFieldGetter(IRFunc* func, IRBitFieldAccessorDecoration*
     // extension behaviour on the right shift
     // return (int(_backing) << (backingWidth-topOfFoo)) >> (backingWidth-fooWidth);
 
-    const auto backingWidth = getIntTypeInfo(backingType).width;
+    const auto backingWidth = maybeGetIntTypeWidth(backingType);
     const auto fieldWidth = dec->getFieldWidth();
     const auto topOfField = dec->getFieldOffset() + fieldWidth;
-    const auto leftShiftAmount = backingWidth - topOfField;
-    const auto rightShiftAmount = backingWidth - fieldWidth;
+
+    auto castBackingType = backingType;
+    if (getIntTypeSigned(backingType) != getIntTypeSigned(func->getResultType()))
+        castBackingType = builder.getType(getOppositeSignIntTypeOp(backingType->getOp()));
```

Comment:
The previous logic using `getIntTypeOpFromInfo` needed to know the width, despite only swapping the sign. This approach does not need the width and works on IntPtr.